packages/server/src/database/migrations/20200212131402-normalize-users.js

```diff
@@ -0,0 +1,127 @@
+const Logger = require('@aragon/court-backend-shared/helpers/logger')
+const subscriptionsMigration = require('./20191219135036-create-subscription')
+const { isAddress, toChecksumAddress } = require('web3-utils')
+
+Logger.setDefaults(false, false)
+const logger = Logger('Migrator')
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    await queryInterface.createTable('Users',
+      {
+        id: {
+          allowNull: false,
+          autoIncrement: true,
+          primaryKey: true,
+          type: Sequelize.INTEGER
+        },
+        email: {
+          type: Sequelize.STRING,
+          allowNull: false,
+          unique: true,
+        },
+        createdAt: {
+          allowNull: false,
+          type: Sequelize.DATE
+        },
+        updatedAt: {
+          allowNull: false,
+          type: Sequelize.DATE
+        }
+      }
+    )
+
+    await queryInterface.createTable('UserAddresses',
+      {
+        id: {
+          allowNull: false,
+          autoIncrement: true,
+          primaryKey: true,
+          type: Sequelize.INTEGER
+        },
+        address: {
+          type: Sequelize.STRING,
+          allowNull: false,
+          unique: true,
+        },
+        userId: {
+          type: Sequelize.INTEGER,
+          references: {
+            model: 'Users',
+            key: 'id',
+            as: 'userId',
+          },
+        },
+        createdAt: {
+          allowNull: false,
+          type: Sequelize.DATE
+        },
+        updatedAt: {
+          allowNull: false,
+          type: Sequelize.DATE
+        }
+      }
+    )
+
+    console.log('')
+    const subscriptions = (await queryInterface.sequelize.query(`SELECT * FROM "Subscriptions" ORDER BY id`))[0]
+
+    for (const { email, address, createdAt, updatedAt } of subscriptions) {
+      const parsedEmail = email.toLowerCase()
+      const existingUser = (await queryInterface.sequelize.query(`SELECT id FROM "Users" WHERE email='${parsedEmail}'`))[0]
+
+      if (existingUser.length === 0) {
+        logger.success(`Inserting new user with email '${parsedEmail}'`)
+        await queryInterface.sequelize.query(`INSERT INTO "Users" ("email", "createdAt", "updatedAt") VALUES ('${parsedEmail}', '${createdAt.toUTCString()}', '${updatedAt.toUTCString()}')`)
+      }
+      else {
+        const { id: userId } = existingUser[0]
+        logger.warn(`Skipping email '${parsedEmail}', user with ID '${userId}' already has that email address`)
+      }
+
+      const { id: userId } = (await queryInterface.sequelize.query(`SELECT id FROM "Users" WHERE email='${parsedEmail}'`))[0][0]
+      if (!address) logger.warn(`Skipping address '${address}' for user with ID '${userId}'`)
+      else {
+        const parsedAddress = isAddress(address) ? toChecksumAddress(address) : address
```

Comment:
I wonder here, should we skip and log and error if the address doesn't parse to a correct checksum address?

Reply:
Definitely, was about to fix that :)